modules/core/shared/src/test/scala/gem/arb/ArbStep.scala

```diff
@@ -5,28 +5,71 @@ package gem
 package arb
 
 import gem.enum.{Instrument, SmartGcalType}
-import gem.config.{GcalConfig, TelescopeConfig}
+import gem.config.{DynamicConfig, GcalConfig, TelescopeConfig}
 import org.scalacheck._
 import org.scalacheck.Arbitrary._
 
 trait ArbStep {
 
   import ArbEnumerated._
   import ArbGcalConfig._
+  import ArbGmos._
   import ArbDynamicConfig._
   import ArbTelescopeConfig._
 
+  import Step.Base._
+
+  implicit val arbStepBaseBias: Arbitrary[Bias.type] =
+    Arbitrary(Gen.const(Bias))
+
+  implicit val cogStepBaseBias: Cogen[Bias.type] =
+    Cogen[Unit].contramap(Function.const(()))
+
+  implicit val arbStepBaseDark: Arbitrary[Dark.type] =
+    Arbitrary(Gen.const(Dark))
+
+  implicit val cogStepBaseDark: Cogen[Dark.type] =
+    Cogen[Unit].contramap(Function.const(()))
+
+  implicit val arbStepBaseGcal: Arbitrary[Gcal] =
+    Arbitrary(arbitrary[GcalConfig].map(Gcal(_)))
+
+  implicit val cogStepBaseGcal: Cogen[Gcal] =
+    Cogen[GcalConfig].contramap(_.gcal)
+
+  implicit val arbStepBaseScience: Arbitrary[Science] =
+    Arbitrary(arbitrary[TelescopeConfig].map(Science(_)))
+
+  implicit val cogStepBaseScience: Cogen[Science] =
+    Cogen[TelescopeConfig].contramap(_.telescope)
+
+  implicit val arbStepBaseSmartGcal: Arbitrary[SmartGcal] =
+    Arbitrary(arbitrary[SmartGcalType].map(SmartGcal(_)))
+
+  implicit val cogStepBaseSmartGcal: Cogen[SmartGcal] =
+    Cogen[SmartGcalType].contramap(_.smartGcalType)
+
   implicit val arbStepBase: Arbitrary[Step.Base] =
     Arbitrary {
       Gen.oneOf(
-        Gen.const(Step.Base.Bias),
-        Gen.const(Step.Base.Dark),
-        arbitrary[GcalConfig].map(Step.Base.Gcal(_)),
-        arbitrary[TelescopeConfig].map(Step.Base.Science(_)),
-        arbitrary[SmartGcalType].map(Step.Base.SmartGcal)
+        arbitrary[Bias.type],
+        arbitrary[Dark.type],
+        arbitrary[Gcal     ],
+        arbitrary[Science  ],
+        arbitrary[SmartGcal]
       )
     }
 
+  implicit val cogStepBase: Cogen[Step.Base] =
+    Cogen[(Option[Bias.type], Option[Dark.type], Option[Gcal], Option[Science], Option[SmartGcal])]
+      .contramap {
+        case Bias         => (Some(Bias), None,       None,    None,    None   )
+        case Dark         => (None,       Some(Dark), None,    None,    None   )
+        case b: Gcal      => (None,       None,       Some(b), None,    None   )
+        case b: Science   => (None,       None,       None,    Some(b), None   )
+        case b: SmartGcal => (None,       None,       None,    None,    Some(b))
+      }
```

Comment:
This feels particularly wrong, but I'm not sure how to write the `Cogen` otherwise.

Reply:
I think I usually do `Either[Bias, Either[Dark, Either[...`

Not that it feels much better